plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt

```diff
@@ -61,11 +61,11 @@ internal class LocationPuckManager(
       animationManager.setLocationLayerRenderer(locationLayerRenderer)
       animationManager.applyPulsingAnimationSettings(settings)
       animationManager.applyAccuracyRadiusSettings(settings2)
-      locationLayerRenderer.addLayers(positionManager)
       lastLocation?.let {
         updateCurrentPosition(it)
       }
       updateCurrentBearing(lastBearing)
+      locationLayerRenderer.addLayers(positionManager)
```

Comment:
so this should be called after `updateCurrentBearing`?

Reply:
Exactly. Otherwise renderer may run render pass before the updateCurrentBearing will be called thus will use incorrect default zero bearing.
The call sequence becomes : 
```
main thread : locationLayerRenderer.addLayers -> layers are added
render thread : render pass -> no bearing value is set, resets to default
main thread : updateCurrentBearing -> will be applied on the next render pass
```

Reply:
This block will only be executed during init but in your screen recording, it's a long-lasting period. Wondering whether this is the root cause for it.

Reply:
Then it makes sense.